packages/snaps-controllers/src/snaps/SnapController.test.ts

```diff
@@ -99,16 +102,34 @@ import {
   SNAP_APPROVAL_UPDATE,
 } from './SnapController';
 
-Object.defineProperty(globalThis, 'crypto', {
-  value: {
-    // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
-    ...require('node:crypto').webcrypto,
-    getRandomValues: jest.fn().mockReturnValue(new Uint32Array(32)),
-  },
-});
+if (!('CryptoKey' in globalThis)) {
+  // We can remove this once we drop Node 18
+  Object.defineProperty(globalThis, 'CryptoKey', {
+    value: webcrypto.CryptoKey,
+  });
+}
+
+globalThis.crypto ??= webcrypto as typeof globalThis.crypto;
+globalThis.crypto.getRandomValues = <Type extends ArrayBufferView | null>(
+  array: Type,
+) => {
+  if (array === null) {
+    return null as Type;
+  }
+
+  return new Uint8Array(array.buffer).fill(0) as unknown as Type;
```

Comment:
Why not `array.fill(0)` and return the original array? IIRC that's similar to what the original function does.

Reply:
https://github.com/MetaMask/snaps/pull/2316#discussion_r1557472998